src/molecules/VoiceRecorder/index.tsx

```diff
@@ -0,0 +1,269 @@
+      recorder.addEventListener('dataavailable', (event) => {
+        audioChunks.push(event.data)
+      })
+
+      //analysis:
+      const audioContext = new AudioContext()
+      const audioStreamSource = audioContext.createMediaStreamSource(stream)
+      const analyser = audioContext.createAnalyser()
+      analyser.minDecibels = VOICE_MIN_DECIBELS
+      audioStreamSource.connect(analyser)
+      const bufferLength = analyser.frequencyBinCount
+      const domainData = new Uint8Array(bufferLength)
+
+      //loop:
+      let time: Date = new Date()
+      let startTime: number
+      let lastDetectedTime: number = time.getTime()
+      let anySoundDetected: boolean = false
+      const detectSound = () => {
+        //recording stopped by user:
+        if (!IS_RECORDING) return
+
+        time = new Date()
+        const currentTime = time.getTime()
+
+        //time out:
+        if (currentTime > startTime + DIALOG_MAX_LENGTH) {
+          recorder.stop()
+          return
+        }
+
+        //a dialog detected:
+        if (
+          anySoundDetected === true &&
+          currentTime > lastDetectedTime + delayBetweenDialogs
+        ) {
+          recorder.stop()
+          return
+        }
+
+        //check for detection:
+        analyser.getByteFrequencyData(domainData)
+        for (let i = 0; i < bufferLength; i++)
+          if (domainData[i] > 0) {
+            anySoundDetected = true
+            time = new Date()
+            lastDetectedTime = time.getTime()
+          }
+
+        //continue the loop:
+        window.requestAnimationFrame(detectSound)
+      }
+      window.requestAnimationFrame(detectSound)
+
+      //stop event:
+      recorder.addEventListener('stop', () => {
+        //stop all the tracks:
+        stream.getTracks().forEach((track) => track.stop())
+        if (!anySoundDetected) return
+
+        //send to server:
+        const audioBlob = new Blob(audioChunks, { type: 'audio/mp3' })
+        makeComputeAPICall(audioBlob)
+      })
+    })
+  }
+  const makeComputeAPICall = async (blob: Blob) => {
+    try {
+      setApiCallStatus('processing')
+      console.log('base', blob)
+      toast.success(`${config.component.messageRecorderWait}`)
+      // Define the API endpoint and make api call here 
+
+      //set api result in setInputMsg 
+      setInputMsg('')
+
+    } catch (error) {
+      console.error(error)
+      setApiCallStatus('error')
+      toast.error(`${config.component.recorderErrorMessage}`)
+      // Set userClickedError to true when an error occurs
+      setUserClickedError(false)
+      setTimeout(() => {
+        // Check if the user has not clicked the error icon again
+        if (!userClickedError) {
+          setApiCallStatus('idle')
+        }
+      }, 2500)
+    
+    }
+  }
+
+  return (
+    <div>
+      <div>
+        {mediaRecorder && mediaRecorder.state === 'recording' ? (
+          <div className={styles.center}>
+            {includeDiv ? (
```

Comment:
don't use that much of nested conditions ,you can check this code - [link](https://github.com/samagra-comms/uci-web-channel/blob/feat-new-ui2/src/components/recorder/RenderVoiceRecorder.jsx)